UnicornEngine/system/source/adapter/glfw/Adapter.cpp

```diff
@@ -0,0 +1,1431 @@
+
+    result.name = glfwGetMonitorName(monitorHandle);
+
+    glfwGetMonitorPhysicalSize(monitorHandle, &result.physicalSize.first, &result.physicalSize.second);
+
+    {
+        int modeCount = 0;
+        const GLFWvidmode* modes = glfwGetVideoModes(monitorHandle, &modeCount);
+
+        result.modes.reserve(modeCount);
+
+        for (int m = 0; m < modeCount; ++m)
+        {
+            result.modes.push_back(PrepareVideoMode(&modes[m]));
+        }
+    }
+
+    result.isPrimary = ( monitorHandle == primary );
+    result.id = static_cast<uint32_t>(reinterpret_cast<uintptr_t>(handle));
+    result.state = MonitorMemento::State::Connected;
+    result.handle = handle;
+
+    return result;
+}
+
+void* Adapter::GetPrimaryMonitor()
+{
+    return glfwGetPrimaryMonitor();
+}
+
+VideoMode Adapter::GetActiveVideoMode(void* handle)
+{
+    return PrepareVideoMode(glfwGetVideoMode(static_cast<GLFWmonitor*>(handle)));
+}
+
+std::pair<int32_t, int32_t> Adapter::GetVirtualPosition(void* handle)
+{
+    std::pair<int32_t, int32_t> result;
+
+    glfwGetMonitorPos(static_cast<GLFWmonitor*>(handle), &result.first, &result.second);
+
+    return result;
+}
+
+GammaRamp Adapter::GetGammaRamp(void* handle)
+{
+    const GLFWgammaramp* ramp = glfwGetGammaRamp(static_cast<GLFWmonitor*>(handle));
+
+    GammaRamp result = {};
+
+    if (ramp)
+    {
+        result.size = ramp->size;
+        result.red = new uint16_t[result.size];
+        result.green = new uint16_t[result.size];
+        result.blue = new uint16_t[result.size];
+
+        for (uint32_t g = 0; g < result.size; ++g)
+        {
+            result.red[g] = ramp->red[g];
+            result.green[g] = ramp->green[g];
+            result.blue[g] = ramp->blue[g];
+        }
+    }
+    else
+    {
+        result = GammaRamp();
+    }
+
+    return result;
+}
+
+void Adapter::SetGammaRamp(void* handle, const GammaRamp& gammaRamp)
+{
+    GLFWgammaramp ramp;
+
+    ramp.size = gammaRamp.size;
+    ramp.red = gammaRamp.red;
+    ramp.green = gammaRamp.green;
+    ramp.blue = gammaRamp.blue;
+
+    glfwSetGammaRamp(static_cast<GLFWmonitor*>(handle), &ramp);
+}
+
+std::vector<void*> Adapter::GetConnectedGamepads()
+{
+    std::vector<void*> result;
+
+    result.reserve(4);
+
+    for (int i = GLFW_JOYSTICK_1; i <= GLFW_JOYSTICK_LAST; ++i)
+    {
+        if (GLFW_TRUE == glfwJoystickPresent(i))
+        {
+            result.push_back(reinterpret_cast<void*>(static_cast<intptr_t>(i)));
+        }
+    }
+
+    return result;
+}
+
+const char* Adapter::GetGamepadName(void* handle)
+{
+    return glfwGetJoystickName(static_cast<int>(reinterpret_cast<intptr_t>(handle)));
+}
+
+std::vector<float> Adapter::GetGamepadAxes(void* handle)
+{
+    const int gamepad = static_cast<int>(reinterpret_cast<intptr_t>(handle));
+    int count = 0;
+
+    const float* axes = glfwGetJoystickAxes(gamepad, &count);
+
+    std::vector<float> result(count);
```

Comment:
Without std::transform it won't be possible due to conversion of values. And I am yet not sure about the std::transform